src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -505,13 +505,40 @@ InterpreterSelectQuery::InterpreterSelectQuery(
     {
         /// Allow push down and other optimizations for VIEW: replace with subquery and rewrite it.
         ASTPtr view_table;
+        NameToNameMap parameter_values;
+        NameToNameMap parameter_types;
         if (view)
-            view->replaceWithSubquery(getSelectQuery(), view_table, metadata_snapshot);
+        {
+            query_info.is_parameterized_view = view->isParameterizedView();
+            /// We need to fetch the parameters set for SELECT parameterized view before the query is replaced.
+            /// replaceWithSubquery replaces the function child and adds the subquery in its place.
+            /// the parameters are children of function child, if function is replaced the parameters are also gone from tree
```

Comment:
correct me if I am wrong 
```suggestion
            /// the parameters are children of function child, if function (which corresponds to parametrised view and has parameters in its arguments: `parametrised_view(<params>)`) is replaced the parameters are also gone from tree
```

Reply:
addressed it